Documentation/auto-dependency-flow/api.md

```diff
@@ -0,0 +1,119 @@
+      https://dotnet.myget.org/F/dotnet-core/api/v3/index.json;
+      https://api.nuget.org/v3/index.json
+    </RestoreSources>
+  </PropertyGroup>
+</Project>
+```
+
+If the repo doesn't use [NuGet MSBuild targets](https://docs.microsoft.com/en-us/nuget/schema/msbuild-targets#restore-target) to restore, there's more work to use the `RestoreSources` property. For example, to pass `--source` args, `RestoreSources` can be `Include`d in an Item element to split it, then formatted.
+
+There is no known reasonable way to use NuGet.Config files, satisfy this API, *and* avoid duplicating the default source information. The recommendation is to not use NuGet.Config.
+
+## `/p:DotNetAssetRootUrl=<url>`
+Directs the repo to get binary assets from a given base `url`. For example, installers and lzma files. The scheme can be `http[s]` or `file`. The url ends with a `/`.
+
+### Conventions
+Repos must cooperate to establish conventions. Core-Setup needs to implement `DotNetOutputBlobFeedDir` in a way that allows CLI to find its binaries in `DotNetAssetRootUrl`.
+
+### Recommended implementation
+Use the AssetRoot properties if they are defined. Example (adapted from [CLI's build/BundledRuntimes.props](https://github.com/dotnet/cli/blob/4fe4c4d28a5171946311ca3ebf65af95180eb11f/build/BundledRuntimes.props)):
+
+```xml
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <!-- SharedFrameworkVersion is defined elsewhere by a Core-Setup package version. -->
+    <CombinedFrameworkHostCompressedFileName>dotnet-runtime-$(SharedFrameworkVersion)-$(SharedFrameworkRid)$(ArchiveExtension)</CombinedFrameworkHostCompressedFileName>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <CoreSetupBlobRootUrl Condition="'$(DotNetAssetRootUrl)' != ''">$(DotNetAssetRootUrl)</CoreSetupBlobRootUrl>
+    <CoreSetupBlobRootUrl Condition="'$(CoreSetupBlobRootUrl)' == ''">https://dotnetcli.azureedge.net/dotnet/</CoreSetupBlobRootUrl>
+
+    <CoreSetupRootUrl>$(CoreSetupBlobRootUrl)Runtime/</CoreSetupRootUrl>
+
+    <CoreSetupBlobAccessTokenParam Condition="'$(DotNetAssetRootAccessTokenSuffix)' != ''">$(DotNetAssetRootAccessTokenSuffix)</CoreSetupBlobAccessTokenParam>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <_DownloadAndExtractItem Include="CombinedSharedHostAndFrameworkArchive">
+      <Url>$(CoreSetupRootUrl)$(SharedFrameworkVersion)/$(CombinedFrameworkHostCompressedFileName)$(CoreSetupBlobAccessTokenParam)</Url>
+    </_DownloadAndExtractItem>
+  </ItemGroup>
+</Project>
+```
+
+The above assumes that Core-Setup publishes outputs to `$(DotNetOutputBlobFeedDir)assets/Runtime/<SharedFrameworkVersion>/`.
+
+## `/p:DotNetAssetRootAccessTokenSuffix=<query string>`
+Directs the repo to append `query string` to any URL constructed using the `DotNetAssetRootUrl`.
+
+Recommended implementation is included in the `DotNetAssetRootUrl` section.
+
+
+# Output placement arguments
+
+## `/p:DotNetOutputBlobFeedDir=<target directory>`
```

Comment:
What does the word `blob` mean in this property?  Is it linked to Azure blobs in some way?

Reply:
This refers to "blob feed", which is the dir structure this API tells repos to put outputs into. It's not a free-for-all dumping ground, so I think it makes sense to have the structure's name in the property rather than something more generic. Blob feeds are briefly described in the terms.

For what "blob" means in "blob feed", I don't know. Sometime soon I'm hoping there will be docs in BuildTools I can link to that will explain.